bluesky/callbacks/broker.py

```diff
@@ -7,7 +7,34 @@
 import matplotlib.pyplot as plt
 
 
-class LiveImage(CallbackBase):
+class BrokerCallbackBase:
+    """
+    Base class for callbacks which need filled documents
+
+    Parameters
+    ----------
+    field: str
+        name of data field in an Event
+    fs: FileStore instance
+        The FileStore instance to pull the data from
+    """
+    def __init__(self, fields, *, fs=None):
+        if not isinstance(fields, (tuple, list)):
+            fields = (fields, )
+        if fs is None:
+            import filestore.api as fs
+        self.fs = fs
+        self.fields = fields
+
+    def event(self, doc):
+        for field in self.fields:
+            if doc.get('filled', {}).get(field):
+                pass
+            else:
+                doc['data'].update(self.fs.retrieve(doc['data'][field]))
```

Comment:
We should really give you a nice way to do this with a Broker instance. Users shouldn't have to juggle a filestore instance, especially since there might be multiple different ones needed to fill an event. Track https://github.com/NSLS-II/databroker/issues/213